dora/core/server/worker/src/main/java/alluxio/worker/dora/DoraOpenFileHandles.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.Constants;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A collection of open file handles in a dora worker.
+ *
+ * It is also a thread, and will run periodic checking of stale open handles.
+ */
+public class DoraOpenFileHandles extends Thread {
+  private final Map<String, OpenFileHandle> mOpenFileHandles;
+  private Boolean mStop;
+
+  DoraOpenFileHandles() {
+    mOpenFileHandles = new HashMap<>();
+    mStop = Boolean.FALSE;
+  }
+
+  @Override
+  public void run() {
+    while (!mStop) {
+      try {
+        sleep(Constants.MINUTE_MS);
+        // Iterate the mOpenFileHandles if some handles are stale (not active for a long time).
+        Set<String> keys = mOpenFileHandles.keySet();
+        for (String key : keys) {
+          OpenFileHandle handle = mOpenFileHandles.get(key);
+          if (System.currentTimeMillis() - handle.getLastAccessTimeMs() >= Constants.HOUR) {
+            mOpenFileHandles.remove(key);
+            handle.close();
```

Comment:
what if any exception occurs in the loop? then the thread would exit and then we cannot release any handle any more. am I right?

Reply:
Added another exception catch.
Then, this thread will not exit on exceptions unless the mStop is `changed` to `true` when shutdown() is called.